lib/db/schema/patch-045-046.sql

```diff
@@ -0,0 +1,188 @@
+-- Add emails table and corresponding procedures to
+-- create, get and delete them.
+
+CREATE TABLE IF NOT EXISTS emails (
+  normalizedEmail VARCHAR(255) NOT NULL PRIMARY KEY,
+  email VARCHAR(255) NOT NULL,
+  uid BINARY(16) NOT NULL,
+  emailCode BINARY(16) NOT NULL,
+  isVerified BOOLEAN NOT NULL DEFAULT FALSE,
+  isPrimary BOOLEAN NOT NULL DEFAULT FALSE,
+  verifiedAt BIGINT UNSIGNED,
+  createdAt BIGINT UNSIGNED NOT NULL
+) ENGINE=InnoDB;
+
+DROP procedure IF EXISTS `createEmail_1`;
+
+CREATE PROCEDURE `createEmail_1` (
+    IN `normalizedEmail` VARCHAR(255),
+    IN `email` VARCHAR(255),
+    IN `uid` BINARY(16) ,
+    IN `emailCode` BINARY(16),
+    IN `isVerified` TINYINT(1),
+    IN `isPrimary` TINYINT(1),
+    IN `verifiedAt` BIGINT UNSIGNED,
+    IN `createdAt` BIGINT UNSIGNED
+)
+BEGIN
+    DECLARE EXIT HANDLER FOR SQLEXCEPTION
+        BEGIN
+        ROLLBACK;
+        RESIGNAL;
+    END;
+
+    START TRANSACTION;
+
+    -- Currently, can not add an email that is specified in the
+    -- accounts table, regardless of verification state.
+    SET @emailExists = 0;
+    SELECT COUNT(*) INTO @emailExists FROM accounts a WHERE a.normalizedEmail = normalizedEmail;
+    IF @emailExists > 0 THEN
+        SIGNAL SQLSTATE '45000' SET MESSAGE_TEXT = 'Email already exists in accounts table.';
+    END IF;
+
+    INSERT INTO emails(
+        normalizedEmail,
+        email,
+        uid,
+        emailCode,
+        isVerified,
+        isPrimary,
+        verifiedAt,
+        createdAt
+    )
+    VALUES(
+        LOWER(normalizedEmail),
+        email,
+        uid,
+        emailCode,
+        isVerified,
+        isPrimary,
+        verifiedAt,
+        createdAt
+    );
+
+    COMMIT;
+END;
+
+CREATE PROCEDURE `accountEmails_1` (
+    IN `inUid` BINARY(16)
+)
+BEGIN
+    -- Return all emails that belong to the uid.
+    -- Email from the accounts table is considered the primary email.
+	(SELECT
+        a.normalizedEmail,
+        a.email,
+        a.uid,
+        a.emailCode,
+        a.emailVerified AS isVerified,
+        TRUE AS isPrimary,
+        a.createdAt AS verifiedAt,
+        a.createdAt
+    FROM
+        accounts a
+    WHERE
+        uid = LOWER(inUid))
+
+    UNION ALL
+
+    (SELECT
+        e.normalizedEmail,
+        e.email,
+        e.uid,
+        e.emailCode,
+        e.isVerified,
+        e.isPrimary,
+        e.verifiedAt,
+        e.createdAt
+    FROM
+        emails e
+    WHERE
+        uid = LOWER(inUid));
+END;
+
+CREATE PROCEDURE `deleteEmail_1` (
+    IN `inNormalizedEmail` VARCHAR(255)
```

Comment:
The caller of this knows both the `uid` and the `email`, and I wonder if it's worth passing both of them in here as a sanity-check.  The query below would become `ELETE FROM emails WHERE normalizedEmail = inNormalizedEmail AND uid = inUid` and it would be an extra line of defense against accidentally deleting an email that doesn't belong to that user.